controllers/vrg_volrep.go

```diff
@@ -1146,61 +1238,102 @@ func (v *VRGInstance) updateVR(volRep *volrep.VolumeReplication,
+	if err != nil {
+		return requeue, false, err
+	}
+
+	if volRep.Spec.ReplicationState == groupState {
+		log.Info("VolumeGroupReplication and VolumeReplicationGroup state match. Proceeding to status check")
+
+		vrStatus := v.checkVRStatus(pvc, volRep, volRep.Status.Conditions, volRep.Status.ObservedGeneration)
+		if vrStatus {
+			v.updatePVCLastSyncTime(pvc.GetNamespace(), pvc.GetName(), volRep.Status.LastGroupSyncTime)
+			v.updatePVCLastSyncDuration(pvc.GetNamespace(), pvc.GetName(), volRep.Status.LastGroupSyncDuration)
+			v.updatePVCLastSyncBytes(pvc.GetNamespace(), pvc.GetName(), volRep.Status.LastGroupSyncBytes)
+		}
+
+		return !requeue, vrStatus, nil
+	}
+
+	volRep.Spec.ReplicationState = groupState
+
+	return v.updateReplicationObject(pvc, volRep, state, log)
+}
+
+func (v *VRGInstance) updateReplicationObject(pvc *corev1.PersistentVolumeClaim, volRep client.Object,
+	state volrep.ReplicationState, log logr.Logger,
+) (bool, bool, error) {
+	const requeue = true
+
 	if err := v.reconciler.Update(v.ctx, volRep); err != nil {
 		log.Error(err, "Failed to update VolumeReplication resource",
-			"name", volRep.Name, "namespace", volRep.Namespace,
+			"name", volRep.GetName(), "namespace", volRep.GetNamespace(),
 			"state", state)
 		rmnutil.ReportIfNotPresent(v.reconciler.eventRecorder, v.instance, corev1.EventTypeWarning,
 			rmnutil.EventReasonVRUpdateFailed, err.Error())
 
 		msg := "Failed to update VolumeReplication resource"
-		v.updatePVCDataReadyCondition(volRep.Namespace, volRep.Name, VRGConditionReasonError, msg)
+		v.updatePVCDataReadyCondition(pvc.GetNamespace(), pvc.GetName(), VRGConditionReasonError, msg)
 
 		return requeue, false, fmt.Errorf("failed to update VolumeReplication resource"+
 			" (%s/%s) as %s, belonging to VolumeReplicationGroup (%s/%s), %w",
-			volRep.Namespace, volRep.Name, state,
+			volRep.GetNamespace(), volRep.GetName(), state,
 			v.instance.Namespace, v.instance.Name, err)
 	}
 
 	log.Info(fmt.Sprintf("Updated VolumeReplication resource (%s/%s) with state %s",
-		volRep.Name, volRep.Namespace, state))
+		volRep.GetName(), volRep.GetNamespace(), state))
 	// Just updated the state of the VolRep. Mark it as progressing.
 	msg := "Updated VolumeReplication resource for PVC"
-	v.updatePVCDataReadyCondition(volRep.Namespace, volRep.Name, VRGConditionReasonProgressing, msg)
+	v.updatePVCDataReadyCondition(pvc.GetNamespace(), pvc.GetName(), VRGConditionReasonProgressing, msg)
 
 	return !requeue, false, nil
 }
 
 // createVR creates a VolumeReplication CR with a PVC as its data source.
-func (v *VRGInstance) createVR(vrNamespacedName types.NamespacedName, state volrep.ReplicationState) error {
-	volumeReplicationClass, err := v.selectVolumeReplicationClass(vrNamespacedName)
+func (v *VRGInstance) createVR(pvc *corev1.PersistentVolumeClaim, state volrep.ReplicationState) error {
+	volumeReplicationClass, err := v.selectVolumeReplicationClass(pvc)
```

Comment:
If an application contains multiple PVCs with different storage class(these PVCs contains a same label that match the VRG label selector). Ramen will create a replicationgroupreplication with a label selector to select all these pvcs, but they have different storage class, will this be a problem for the replicationgroupreplication?